spec/models/miq_schedule_spec.rb

```diff
@@ -105,6 +141,17 @@
         expect(miq_schedule_array['filter']).to be_kind_of(MiqExpression)
         expect(miq_schedule_array['resource_type']).to eq("MiqReport")
       end
+
+      it "imports schedule" do
+        miq_schedule_array = MiqSchedule.export_to_array([miq_schedule.id], MiqSchedule).first["MiqSchedule"]
+        imported_schedule = MiqSchedule.import_from_hash(miq_schedule_array).first
+
+        expect(imported_schedule.sched_action[:method]).to eq("run_report")
+        expect(imported_schedule.userid).to eq("system")
+        expect(imported_schedule.file_depot_id).to eq(file_depot.id)
+        expect(imported_schedule.resource_type).to eq("MiqReport")
```

Comment:
@lpichler I don't think this should be MiqReport though, right?

Reply:
Or rather I don't think 149 here should be ```run_report```

Reply:
yes there is how schedule for database backup looks like:
```
#<MiqSchedule:0x00007f86571a2f40> {
               :id => 153,
             :name => "a",
      :description => "a",
     :sched_action => {
        :method => "db_backup"
    },
           :filter => nil,
    :resource_type => "DatabaseBackup",
           :run_at => {
                :tz => "UTC",
        :start_time => 2019-07-19 00:00:00 UTC,
          :interval => {
             :unit => "once",
            :value => ""
        }
    },
          :enabled => true,
           :userid => "admin",
     :prod_default => nil,
      :last_run_on => nil,
       :created_on => Fri, 19 Jul 2019 10:16:02 UTC +00:00,
       :updated_at => Fri, 19 Jul 2019 10:16:02 UTC +00:00,
    :miq_search_id => nil,
          :zone_id => 1,
            :adhoc => nil,
    :file_depot_id => 9,
      :resource_id => nil
}

```

Reply:
isn't that what https://github.com/ManageIQ/manageiq/pull/19257/files#diff-1b41ae2e55bcc21aa5203e5650408cd2R130 is doing?

Reply:
did you want to refer specific line ? if so it is now working :/

Reply:
Yeah, I did want to refer to specific line. I'm confused about what this test is doing wrong though.

Reply:
It's the same setup that these specs use for context "valid schedules for db_backup" do. What am I missing?

Reply:
ah I see it now, but you need to put it to `let(:miq_schedule)` because `DatabaseBackup` schedule - is created and in `@schedule` variable.

so 

instead of 
```
     let(:miq_schedule) do
        FactoryBot.create(:miq_schedule, :updated_at => 1.year.ago, :filter => miq_expression, :sched_action => sched_action, :userid => user.userid, :last_run_on => Time.zone.now, :file_depot => file_depot)
      end
```

you need object what are you creating on line 130

```
FactoryBot.create(:miq_schedule_validation, :run_at => run_at, :file_depot => file_depot, :sched_action => {:method => "db_backup"}, :resource_type => "DatabaseBackup")
```

then the related before block is more then we want and it looks like we can define our `DatabaseBackup` schedule - without it.